internal/services/watch_btc_events.go

```diff
@@ -0,0 +1,589 @@
+	defer cancel()
+
+	// Get spending details
+	select {
+	case spendDetail := <-spendEvent.Spend:
+		log.Debug().
+			Str("staking_tx", delegation.StakingTxHashHex).
+			Str("unbonding_tx", spendDetail.SpendingTx.TxHash().String()).
+			Msg("unbonding tx has been spent")
+		if err := s.handleSpendingUnbondingTransaction(
+			quitCtx,
+			spendDetail.SpendingTx,
+			spendDetail.SpenderInputIndex,
+			delegation,
+		); err != nil {
+			log.Error().
+				Err(err).
+				Str("staking_tx", delegation.StakingTxHashHex).
+				Str("unbonding_tx", spendDetail.SpendingTx.TxHash().String()).
+				Msg("failed to handle spending unbonding transaction")
+			return
+		}
+
+	case <-s.quit:
+		return
+	case <-quitCtx.Done():
+		return
+	}
+}
+
+func (s *Service) handleSpendingStakingTransaction(
+	ctx context.Context,
+	spendingTx *wire.MsgTx,
+	spendingHeight,
+	spendingInputIdx uint32,
+	stakingTxHashHex string,
+) error {
+	delegation, err := s.db.GetBTCDelegationByStakingTxHash(ctx, stakingTxHashHex)
+	if err != nil {
+		return fmt.Errorf("failed to get BTC delegation by staking tx hash: %w", err)
+	}
+
+	paramsVersion := s.GetVersionedGlobalParamsByHeight(delegation.StakingTx.StartHeight)
+	if paramsVersion == nil {
+		log.Ctx(ctx).Error().Msg("failed to get global params")
+		return types.NewErrorWithMsg(
+			http.StatusInternalServerError, types.InternalServiceError,
+			"failed to get global params based on the staking tx height",
+		)
+	}
+
+	// First try to validate as unbonding tx
+	isUnbonding, err := s.IsValidUnbondingTx(spendingTx, delegation, paramsVersion)
+	if err != nil {
+		if errors.Is(err, types.ErrInvalidUnbondingTx) {
+			metrics.IncrementInvalidUnbondingTxCounter()
+			log.Error().
+				Err(err).
+				Str("staking_tx", delegation.StakingTxHashHex).
+				Msg("found an invalid unbonding tx")
+
+			return nil
+		}
+
+		metrics.IncrementFailedVerifyingUnbondingTxCounter()
+		return fmt.Errorf("failed to validate unbonding tx: %w", err)
+	}
+	if isUnbonding {
+		log.Debug().
+			Str("staking_tx", delegation.StakingTxHashHex).
+			Str("unbonding_tx", spendingTx.TxHash().String()).
+			Msg("staking tx has been spent through unbonding path")
+
+		unbondingTxHex, err := utils.SerializeBtcTransaction(spendingTx)
+		if err != nil {
+			return fmt.Errorf("failed to serialize unbonding tx: %w", err)
+		}
+
+		unbondingTxTimestamp, err := s.btc.GetBlockTimestamp(uint64(spendingHeight))
+		if err != nil {
+			return fmt.Errorf("failed to get block timestamp: %w", err)
+		}
+
+		unbondingEvent := types.NewUnbondingDelegationEvent(
+			delegation.StakingTxHashHex,
+			uint64(spendingHeight),
+			unbondingTxTimestamp,
+			paramsVersion.UnbondingTime,
+			// valid unbonding tx always has one output
+			uint64(0),
+			unbondingTxHex,
+			spendingTx.TxHash().String(),
+		)
+		utils.PushOrQuit(s.unbondingDelegationChan, unbondingEvent, s.quit)
+
+		// Register unbonding spend notification
+		return s.registerUnbondingSpendNotification(delegation)
```

Comment:
I don't think we should pass `delegation` but we should use unbondingTxHex instead. If I remember correctly the unbonding tx of the delegation is filled when the phase-1 indexer observes it. So it will be empty here